.ci/Jenkinsfile

```diff
@@ -69,7 +69,12 @@ pipeline {
                 unstash 'source'
                 dir("${BASE_DIR}"){
                   withGoEnv(){
-                    goTestJUnit(options: '-v ./...', output: 'junit-report.xml')
+                    whenTrue(isDarwin()) {
```

Comment:
Can you add a comment explaining why there is a Darwin special case here? Otherwise LGTM.

Reply:
```suggestion
                    // goTestSum for go1.18.5 on MacOS produces a segmentation violation. See https://github.com/elastic/elastic-agent-libs/pull/80
                    // For the time being there will not junit transformation in MacOS
                    whenTrue(isDarwin()) {
```

Reply:
Thanks!